File: solutions/go/isbn-verifier/1/isbn_verifier.go
package isbn

import "strings"

func IsValidISBN(isbn string) bool {
	isbn = strings.ReplaceAll(isbn, "-", "")
	if len(isbn) != 10 {
		return false
	}

	sum := 0
	for i, r := range isbn {
		if i == 9 && r == 'X' {
			sum += 10
			continue
		}
		if r >= '0' && r <= '9' {
			sum += (int(r) - '0') * (10 - i)
		} else {
			return false
		}
	}

	return sum%11 == 0
}

File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    isbn = isbn.replace('-','')
    if len(isbn) != 10:
        return False
    sum = 0
    i = 10
    for n in isbn[:9]:
        if not n.isnumeric():
            return False
        sum += int(n) * i
        i -= 1
    if isbn[9] == 'X':
        sum += 10
    elif isbn[9].isnumeric():
            sum += int(isbn[9])
    else: 
        return False
    return sum % 11 == 0